crypto/ocsp/ocsp_server.c

```diff
@@ -70,3 +70,149 @@ int OCSP_basic_add1_cert(OCSP_BASICRESP *resp, X509 *cert) {
+    goto err;
+  }
+
+  // Init |resp->tbsResponseData->responses| if NULL.
+  if (resp->tbsResponseData->responses == NULL) {
+    resp->tbsResponseData->responses = sk_OCSP_SINGLERESP_new_null();
+    if (resp->tbsResponseData->responses == NULL) {
+      goto err;
+    }
+  }
+
+  if (!ASN1_TIME_to_generalizedtime(this_update, &single->thisUpdate)) {
+    goto err;
+  }
+  if (next_update != NULL) {
+    // |next_update| is allowed to be NULL. Only set |single->nextUpdate| if
+    // |next_update| is non-NULL.
+    if (!ASN1_TIME_to_generalizedtime(next_update, &single->nextUpdate)) {
+      goto err;
+    }
+  }
+
+  // Reset |single->certId|.
+  OCSP_CERTID_free(single->certId);
+  single->certId = OCSP_CERTID_dup(cid);
+  if (single->certId == NULL) {
+    goto err;
+  }
+
+  single->certStatus->type = status;
+  switch (single->certStatus->type) {
+    case V_OCSP_CERTSTATUS_REVOKED:
+      if (revoked_time == NULL) {
+        OPENSSL_PUT_ERROR(OCSP, OCSP_R_NO_REVOKED_TIME);
+        goto err;
+      }
+
+      single->certStatus->value.revoked = OCSP_REVOKEDINFO_new();
+      if (single->certStatus->value.revoked == NULL) {
+        goto err;
+      }
+
+      // Start assigning values to |info| once initialized successfully.
+      OCSP_REVOKEDINFO *info = single->certStatus->value.revoked;
+      if (!ASN1_TIME_to_generalizedtime(revoked_time, &info->revocationTime)) {
+        goto err;
+      }
+      // https://www.rfc-editor.org/rfc/rfc5280#section-5.3.1 specifies the only
+      // valid reason codes are 0-10. Value 7 is not used.
+      if (revoked_reason < OCSP_REVOKED_STATUS_UNSPECIFIED ||
+          revoked_reason > OCSP_REVOKED_STATUS_AACOMPROMISE ||
+          revoked_reason == OCSP_UNASSIGNED_REVOKED_STATUS) {
+        OPENSSL_PUT_ERROR(OCSP, OCSP_R_UNKNOWN_FIELD_VALUE);
+        goto err;
+      }
+      info->revocationReason = ASN1_ENUMERATED_new();
+      if (info->revocationReason == NULL ||
+          !ASN1_ENUMERATED_set(info->revocationReason, revoked_reason)) {
+        goto err;
+      }
+
+      break;
+
+    case V_OCSP_CERTSTATUS_GOOD:
+      single->certStatus->value.good = ASN1_NULL_new();
+      if (single->certStatus->value.good == NULL) {
+        goto err;
+      }
+      break;
+
+    case V_OCSP_CERTSTATUS_UNKNOWN:
+      single->certStatus->value.unknown = ASN1_NULL_new();
+      if (single->certStatus->value.unknown == NULL) {
+        goto err;
+      }
+      break;
+
+    default:
+      goto err;
+  }
+
+  // Finally add the |OCSP_SINGLERESP| we were working with to |resp|.
+  if (!sk_OCSP_SINGLERESP_push(resp->tbsResponseData->responses, single)) {
+    goto err;
+  }
+  return single;
+
+err:
+  OCSP_SINGLERESP_free(single);
+  return NULL;
+}
+
+OCSP_RESPONSE *OCSP_response_create(int status, OCSP_BASICRESP *bs) {
+  if (status < OCSP_RESPONSE_STATUS_SUCCESSFUL ||
+      status > OCSP_RESPONSE_STATUS_UNAUTHORIZED ||
+      // 4 is not a valid response status code.
+      status == 4) {
```

Comment:
another magic value

Reply:
I remember this but had thought this was part of another PR haha.. Fixed!